azure/scope/machinepool.go

```diff
@@ -377,11 +377,11 @@ func (m *MachinePoolScope) createMachine(ctx context.Context, machine azure.VMSS
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "scope.MachinePoolScope.createMachine")
 	defer done()
 
-	parsed, err := azureautorest.ParseResourceID(machine.ID)
+	parsed, err := arm.ParseResourceID(machine.ID)
```

Comment:
Are we 100% confident these two `ParseResourceID` interfaces are identical?

Reply:
The implementations are very different (see [go-autorest](https://github.com/Azure/go-autorest/blob/553a90ae65a6a2b18306fa04d7b1625960c5decb/autorest/azure/azure.go#L213) and [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/internal/resource/resource_identifier.go#L63)), but the interface (API) and overall behavior appear to be the same–I tested with some resourceID strings. The SDKv2 `arm` version returns a different struct, but with the same information (although some field names changed).

Reply:
It seems like we don't have existing UT cases for these transformations, so I'll defer to your validations, thank you for doing that work!